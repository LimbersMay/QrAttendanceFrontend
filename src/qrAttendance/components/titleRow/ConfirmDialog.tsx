// ConfirmDialog.jsx
// material ui
import {
    Dialog,
    DialogTitle,
    DialogContent,
    DialogActions,
    Button,
    Box,
    IconButton,
    Typography,
} from '@mui/material';
import { Close } from '@mui/icons-material';

const ConfirmDialog = ({ onCancel, onConfirm }: {onCancel: any, onConfirm: any}) => {
    return (
        <Dialog open={true} maxWidth="sm" fullWidth>
            <DialogTitle> Confirm</DialogTitle>
            <Box position="absolute" top={0} right={0}>
                <IconButton>
                    <Close />
                </IconButton>
            </Box>
            <DialogContent>
                <Typography>Are you sure you want to do delete this?</Typography>
            </DialogContent>
            <DialogActions>
                <Button onClick={onCancel} color="primary" variant="contained">
                    Cancel
                </Button>
                <Button onClick={onConfirm} color="secondary" variant="contained">
                    Confirm
                </Button>
            </DialogActions>
        </Dialog>
    );
};

export default ConfirmDialog;