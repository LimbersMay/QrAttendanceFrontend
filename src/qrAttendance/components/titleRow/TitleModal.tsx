import {ModalLayout} from "../ModalLayout";
import {useUiSlice} from "../../../hooks/useUiSlice";
import {Divider, Grid, TextField} from "@mui/material";
import Typography from "@mui/material/Typography";
import {FormEvent} from "react";
import {useGroupSlice} from "../../../hooks/useGroupSlice";
import {useForm} from "../../../hooks/useForm";
import {QrCodeDatePicker} from "../qrCodeRow/QrCodeDatePicker";
import Button from "@mui/material/Button";

const initialState = {
    id: "",
    name: "",
    date: "2023-02-18T02:19:01.000Z",
}

export const TitleModal = () => {

    const { isTitleModalOpen, toggleTitleModal } = useUiSlice();
    const { active, updateGroup, handleSetActiveGroup } = useGroupSlice();

    const { formState, onInputChange, onDateChange } = useForm(active || initialState);
    const { name, date } = formState;

    const onSubmit = (event: FormEvent) => {
        event.preventDefault();

        if (!active) return;

        updateGroup({
            ...active,
            name,
            date
        });

        handleSetActiveGroup(formState);
        toggleTitleModal();
    }

    return (
        <ModalLayout condition={isTitleModalOpen} handleClose={toggleTitleModal}>
            <form onSubmit={onSubmit}>
                <Grid container>
                    <Grid item xs={12} sx={{textAlign: "center"}}>
                        <Typography variant="h6">Editing {active?.name}</Typography>
                        <Divider />
                    </Grid>

                    <Grid item xs={12} sx={{mt: 2}}>
                        <QrCodeDatePicker date={date} onChangeDate={(date: string) => onDateChange('date', date)}/>
                    </Grid>

                    <Grid item xs={12} sx={{mt: 2}}>
                        <TextField
                            label="Name"
                            name="name"
                            value={name}
                            fullWidth
                            variant={"outlined"}
                            placeholder="name"
                            onChange={onInputChange}>
                        </TextField>
                    </Grid>

                    <Grid item xs={12} sx={{mt: 2}}>
                        <Button
                            variant="contained"
                            fullWidth
                            type={"submit"}
                        >
                            Save
                        </Button>
                    </Grid>
                </Grid>
            </form>
        </ModalLayout>
    )
}
