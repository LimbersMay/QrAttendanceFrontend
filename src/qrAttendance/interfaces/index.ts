export * from "./QrCode";
export * from "./Registry";
