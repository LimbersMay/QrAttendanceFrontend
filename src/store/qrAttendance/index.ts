export * from "./group";
export * from "./qrCode";
export * from "./registry";
