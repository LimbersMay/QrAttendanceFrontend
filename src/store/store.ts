import {AnyAction, configureStore, ThunkAction} from '@reduxjs/toolkit';
import {authSlice} from "./auth";
import {groupSlice, qrCodeSlice, registrySlice} from "./qrAttendance";
import socketMiddleware from "./middlewares/socketMiddleware";
export const store = configureStore({
    reducer: {
        auth: authSlice.reducer,
        group: groupSlice.reducer,
        qrCode: qrCodeSlice.reducer,
        registry: registrySlice.reducer
    },
    middleware: [socketMiddleware]
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<
    ReturnType,
    RootState,
    unknown,
    AnyAction
>
