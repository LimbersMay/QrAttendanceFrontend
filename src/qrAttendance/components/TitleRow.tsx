import React, {useMemo, useState} from "react";

import {IconButton, TableCell, tableCellClasses, TableRow, TextField} from "@mui/material";
import {AddOutlined, SearchOutlined, EditOutlined, SaveOutlined, DeleteOutlined} from "@mui/icons-material";
import {useForm} from "../../hooks/useForm";

import {ConditionalTextField} from "./ConditionalTextField";
import {Group, QrCode} from "../interfaces";
import {useAppDispatch} from "../../store";
import {startNewQrCode, startUpdateGroup} from "../../store/qrAttendance";

export const TitleRow = React.memo(({ group }: {group: Group}) => {

    const initialStateForm = useMemo(() => ({
        groupTitle: group.name,
    }), [group]);

    const dispatch = useAppDispatch();

    const { formState, onInputChange } = useForm(initialStateForm);
    const [isRowEditing, setIsRowEditing] = useState<boolean>(false);

    const { groupTitle } = formState;

    const handleEditRow = () => {
        setIsRowEditing(true);
    }

    const handleSaveRow = () => {
        setIsRowEditing(false);
        dispatch(startUpdateGroup({ ...group, name: groupTitle}));
    }

    const handleDeleteRow = () => {
        console.log('delete group');
    }

    const handleAddEmptyRow = () => {

        const newQrCode: QrCode = {
            id: new Date().toUTCString() + '1',
            groupId: group.id,
            name: 'Default',
            date: '2023-01-18',
            registries: 0,
            enabled: false
        }

        dispatch(startNewQrCode(newQrCode));
    }

    return (
        <TableRow
            sx={{
                [`& .${tableCellClasses.root}`]: {
                    borderBottom: "none"
                }
            }}
        >
            <TableCell />
            <TableCell
                sx={{
                    // TODO: Make this cell look like a title
                    fontSize: "1.4rem",
                }}
                colSpan={2}
            >
                <ConditionalTextField
                    onChange={onInputChange}
                    condition={isRowEditing}
                    name={"groupTitle"}
                    value={groupTitle}
                />
            </TableCell>

            {/* TODO: Add a search field */}
            <TableCell align="right" colSpan={2}>
                <TextField
                    id="outlined-search"
                    label="Search"
                    type="search"
                    variant="outlined"
                    size="small"
                    InputProps={{
                        endAdornment: (
                            <IconButton>
                                <SearchOutlined />
                            </IconButton>
                        )
                    }}
                />
                <IconButton onClick={handleAddEmptyRow}>
                    <AddOutlined/>
                </IconButton>
                {
                    isRowEditing
                     ? <IconButton onClick={handleSaveRow}> <SaveOutlined /> </IconButton>
                     : <IconButton onClick={handleEditRow}> <EditOutlined /> </IconButton>
                }
                <IconButton onClick={handleDeleteRow}>
                    <DeleteOutlined/>
                </IconButton>
            </TableCell>

        </TableRow>
    )
});