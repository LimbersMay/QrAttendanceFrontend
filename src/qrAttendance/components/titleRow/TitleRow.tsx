import React, {useMemo, useState} from "react";

import {IconButton, TableCell, tableCellClasses, TableRow, TextField, Tooltip} from "@mui/material";
import {AddOutlined, SearchOutlined, EditOutlined, SaveOutlined, DeleteOutlined} from "@mui/icons-material";
import {useForm} from "../../../hooks/useForm";

import {ConditionalTextField} from "../ConditionalTextField";
import {Group, QrCode} from "../../interfaces";
import {useAppDispatch} from "../../../store";
import {
    startDeleteGroupWithDependencies,
    startNewQrCode,
    startUpdateGroup
} from "../../../store/qrAttendance";
import ConfirmDialog from "./ConfirmDialog";
import {SnackbarUtilities} from "../../../utilities/snackbar-manager";

export const TitleRow = React.memo(({ group, qrCodes }: {group: Group, qrCodes: QrCode[]}) => {

    const initialStateForm = useMemo(() => ({
        groupTitle: group.name,
    }), [group]);

    const dispatch = useAppDispatch();

    const { formState, onInputChange } = useForm(initialStateForm);

    const [isRowEditing, setIsRowEditing] = useState<boolean>(false);
    const [isTryingToDelete, setIsTryingToDelete] = useState<boolean>(false);

    const { groupTitle } = formState;

    const handleEditRow = () => {
        setIsRowEditing(true);
    }

    const handleToggleDeleteDialog = () => {
        setIsTryingToDelete(!isTryingToDelete);
    }

    const handleSaveRow = () => {
        setIsRowEditing(false);
        dispatch(startUpdateGroup({ ...group, name: groupTitle}));
    }

    const handleDeleteRow = () => {
        dispatch(startDeleteGroupWithDependencies(group.id, qrCodes));
        SnackbarUtilities.sucess(`Group deleted successfully`);
    }

    const handleAddEmptyRow = () => {
        dispatch(startNewQrCode(group.id));
    }

    return (
        <TableRow
            sx={{
                [`& .${tableCellClasses.root}`]: {
                    borderBottom: "none"
                }
            }}
        >
            <TableCell />
            <TableCell
                sx={{
                    // TODO: Make this cell look like a title
                    fontSize: "1.4rem",
                }}
                colSpan={2}
            >
                <ConditionalTextField
                    onChange={onInputChange}
                    condition={isRowEditing}
                    name={"groupTitle"}
                    value={groupTitle}
                />
            </TableCell>

            {/* TODO: Add a search field */}
            <TableCell align="right" colSpan={2}>
                <TextField
                    id="outlined-search"
                    label="Search"
                    type="search"
                    variant="outlined"
                    size="small"
                    InputProps={{
                        endAdornment: (
                            <IconButton>
                                <SearchOutlined />
                            </IconButton>
                        )
                    }}
                />
                <Tooltip title={'Add new qr code'} >
                    <IconButton onClick={handleAddEmptyRow}>
                        <AddOutlined/>
                    </IconButton>
                </Tooltip>

                {
                    isRowEditing
                     ? <IconButton onClick={handleSaveRow}> <SaveOutlined /> </IconButton>
                     : <Tooltip title={'Edit title'} ><IconButton onClick={handleEditRow}> <EditOutlined /> </IconButton></Tooltip>
                }
                <Tooltip title={'Delete group'}>
                    <IconButton onClick={handleToggleDeleteDialog}>
                        <DeleteOutlined/>
                    </IconButton>
                </Tooltip>

                {
                    isTryingToDelete && <ConfirmDialog onConfirm={handleDeleteRow} onCancel={handleToggleDeleteDialog} />
                }

            </TableCell>

        </TableRow>
    )
});