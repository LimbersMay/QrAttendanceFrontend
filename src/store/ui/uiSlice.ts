import {createSlice} from '@reduxjs/toolkit';
import {RootState} from "../store";

interface uiInitialState {
    isQrCodeModalOpen: boolean;
    isRegistryModalOpen: boolean;
    isShowingQrCode: boolean;
    isTitleModalOpen: boolean;
}

const initialState: uiInitialState = {
    isQrCodeModalOpen: false,
    isRegistryModalOpen: false,
    isShowingQrCode: false,
    isTitleModalOpen: false
}

export const uiSlice = createSlice({
    name: 'ui',
    initialState,
    reducers: {
        onShowQrCode: (state) => {
            state.isShowingQrCode = true;
        },
        onHideQrCode: (state) => {
            state.isShowingQrCode = false;
        },
        onToggleQrCodeModal: (state) => {
            state.isQrCodeModalOpen = !state.isQrCodeModalOpen;
        },
        onToggleRegistryModal: (state) => {
            state.isRegistryModalOpen = !state.isRegistryModalOpen;
        },
        onToggleTitleModal: (state) => {
            state.isTitleModalOpen = !state.isTitleModalOpen;
        }
    }
});


// Action creators are generated for each case reducer function
export const { onToggleQrCodeModal, onShowQrCode, onHideQrCode, onToggleRegistryModal, onToggleTitleModal } = uiSlice.actions;
export const selectUi = (state: RootState) => state.ui;