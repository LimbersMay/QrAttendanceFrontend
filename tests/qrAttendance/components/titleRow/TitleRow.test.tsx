import {Provider} from "react-redux";
import {fireEvent, render, screen} from "@testing-library/react";
import {configureStore} from "@reduxjs/toolkit";
import {groups, withActiveGroupAndGroupsState, withQrCodesState} from "../../../fixtures/qrAttendanceStates";
import {TitleRow} from "../../../../src/qrAttendance/components/titleRow/TitleRow";
import {uiSlice} from "../../../../src/store/ui/uiSlice";
import {groupSlice, qrCodeSlice} from "../../../../src/store/qrAttendance";
import {withQrCodeModalOpenTrue} from "../../../fixtures/uiStates";

const store = configureStore({
    reducer: {
        ui: uiSlice.reducer,
        qrCode: qrCodeSlice.reducer,
        group: groupSlice.reducer

    },
    preloadedState: {
        ui: {...withQrCodeModalOpenTrue},
        qrCode: {...withQrCodesState},
        group: {...withActiveGroupAndGroupsState}
    }
});

const mockToggleTitleModal = jest.fn();
const mockToggleQrCodeModal = jest.fn();

const mockStartDeleteGroupWithDependencies = jest.fn();

jest.mock('../../../../src/hooks/useUiStore', () => ({
    useUiStore: () => ({
        toggleTitleModal: mockToggleTitleModal,
        toggleQrCodeModal: mockToggleQrCodeModal,
        isTitleModalOpen: false,
        isQrCodeModalOpen: false,
    })
}));

jest.mock('../../../../src/hooks/useQrAttendanceStore', () => ({
    useQrAttendanceStore: () => ({
        startDeleteGroupWithDependencies: mockStartDeleteGroupWithDependencies
    })
}));

describe('tests for <TitleRow />', () => {

    beforeEach(() => jest.clearAllMocks());

    test('should match the snapshot', () => {

        const {container} = render(
            <Provider store={store}>
                <table>
                    <tbody>
                    <TitleRow group={withActiveGroupAndGroupsState.active || groups[0]}
                              qrCodes={withQrCodesState.qrCodes}/>
                    </tbody>
                </table>
            </Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('should call the toggleQrCodeModal when add button is clicked', () => {

        render(
            <Provider store={store}>
                <table>
                    <tbody>
                    <TitleRow group={withActiveGroupAndGroupsState.active || groups[0]}
                              qrCodes={withQrCodesState.qrCodes}/>
                    </tbody>
                </table>
            </Provider>
        );

        const addButton = screen.getByLabelText('addQrCodeButton');
        fireEvent.click(addButton);

        expect(mockToggleQrCodeModal).toHaveBeenCalled();
    });

    test('should call the toggleTitleModal function when the edit button is clicked', () => {

        render(
            <Provider store={store}>
                <table>
                    <tbody>
                    <TitleRow group={withActiveGroupAndGroupsState.active || groups[0]}
                              qrCodes={withQrCodesState.qrCodes}/>
                    </tbody>
                </table>
            </Provider>
        );

        const editButton = screen.getByLabelText('editGroupButton');
        fireEvent.click(editButton);

        expect(mockToggleTitleModal).toHaveBeenCalled();
    });

    test('should show the confirm dialog when the delete button is clicked', () => {

        render(
            <Provider store={store}>
                <table>
                    <tbody>
                    <TitleRow group={withActiveGroupAndGroupsState.active || groups[0]}
                              qrCodes={withQrCodesState.qrCodes}/>
                    </tbody>
                </table>
            </Provider>
        );

        const deleteButton = screen.getByLabelText('deleteGroupButton');
        fireEvent.click(deleteButton);

        const {rerender} = render(
            <Provider store={store}>
                <table>
                    <tbody>
                    <TitleRow group={withActiveGroupAndGroupsState.active || groups[0]}
                                qrCodes={withQrCodesState.qrCodes}/>
                    </tbody>
                </table>
            </Provider>
        );

        // rerender to show the confirm dialog
        rerender(
            <Provider store={store}>
                <table>
                    <tbody>
                    <TitleRow group={withActiveGroupAndGroupsState.active || groups[0]}
                                qrCodes={withQrCodesState.qrCodes}/>
                    </tbody>
                </table>
            </Provider>
        );

        const confirmTitle = screen.getByLabelText('Confirm');
        expect(confirmTitle).toBeTruthy();
    });
});
