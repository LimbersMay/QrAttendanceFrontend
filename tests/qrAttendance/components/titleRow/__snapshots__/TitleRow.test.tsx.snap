// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <TitleRow /> should match the snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="MuiTableRow-root css-1xupr2d-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
        />
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-zqz43s-MuiTableCell-root"
          colspan="2"
        >
          System engineering
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeMedium css-r7oxtk-MuiTableCell-root"
          colspan="2"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-k7ysl6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-search"
              id="outlined-search-label"
            >
              Search
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-vvf66v-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-search"
                type="search"
                value=""
              />
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Search
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            aria-label="addQrCodeButton"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="editGroupButton"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
             
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EditOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m14.06 9.02.92.92L5.92 19H5v-.92l9.06-9.06M17.66 3c-.25 0-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.2-.2-.45-.29-.71-.29zm-3.6 3.19L3 17.25V21h3.75L17.81 9.94l-3.75-3.75z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="deleteGroupButton"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 9v10H8V9h8m-1.5-6h-5l-1 1H5v2h14V4h-3.5l-1-1zM18 7H6v12c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
